### DISCUSSION TITLE
Manual Setup - Problems

### DISCUSSION THREAD

**Discussion by fuadar** [jasonacox/Powerwall-Dashboard#184] [Q&A]
Have installed powerwall-dashboard using manual steps. All looks good but my graphs only have weather data i can see the grid flow from the powerwall but no other grid data 
docker logs -f looks clean as well
Here is an example of my Grafana dashboard 
I've also run the pypowerwall curls and i can  see data is coming 
![image](https://user-images.githubusercontent.com/2882496/220005313-d7b7fe6d-9799-453c-9672-f665f2e67b35.png)

**Comment by jasonacox** [comment]
Hi @fuadar !  Thanks for opening this. I confess I haven't tested the "manual steps" in quite a while but they should mirror what is in `setup.sh`.

What we know:
* If you are able to see anything on the dashboard, you have `grafana` running. :)
* If you are able to see the animation, that means `pypowerwall` is running correctly.
* If you see weather data on the graphs, that means `weather411` and `influxdb` are running.
* We have no indication that the InfluxDB continuous queries (CQs) are running to build the display data. 
* We have no indication that `telegraf` is running - this tools copies data from pypowerwall into influxdb

## InfluxDB CQs

You can run this again - it won't hurt even if you already ran it during the manual steps:

```bash
  docker exec -it influxdb influx -import -path=/var/lib/influxdb/influxdb.sql
```
Does it report any errors?  Give it 5m and check the dashboard (reload).

## Telegraf

This is a required tool to get the data into InfluxDB.  Check it:

```bash
# check telegraf for logs
docker logs telegraf

# check to make sure telegraf config files are there
ls -l telegraf*
```

```
-rw-r--r-- 1 pi pi 212602 Feb 19 15:54 telegraf.conf
-rw-r--r-- 1 pi pi 212602 Feb 19 15:54 telegraf.conf.bak
-rw-r--r-- 1 pi pi   1818 Feb 19 15:54 telegraf.local
-rw-r--r-- 1 pi pi   1818 Feb 19 15:54 telegraf.local.sample
```

The important ones are `telegraf.conf` and `telegraf.local`.  Check both files to see if the content looks correct.  Key section in telegraf.local:

```python
[[outputs.influxdb]]
	urls = ["http://influxdb:8086"]
	database = "powerwall"
	skip_database_creation = false
	retention_policy = "raw"

[[inputs.http]]
	urls = [
		"http://pypowerwall:8675/aggregates",
		"http://pypowerwall:8675/soe",
		"http://pypowerwall:8675/strings",
		"http://pypowerwall:8675/temps/pw",
		"http://pypowerwall:8675/freq",
		"http://pypowerwall:8675/pod"
	]
	method = "GET"
	insecure_skip_verify = true
	timeout = "4s"
	data_format = "json"

[[processors.date]]
	tag_key = "month"
	date_format = "Jan"
	timezone = "America/Los_Angeles"

[[processors.date]]
	tag_key = "year"
	date_format = "2006"
	timezone = "America/Los_Angeles"

# Alert Data
[[inputs.http]]
	urls = [
		"http://pypowerwall:8675/alerts/pw"
	]
	name_override = "alerts"
	method = "GET"
	insecure_skip_verify = true
	timeout = "4s"
	data_format = "json"
```

**Comment by fuadar** [comment]
Issue does seem to be in telgraf 
I see https://github.com/jasonacox/Powerwall-Dashboard/blob/main/telegraf.local.sample
All lines are commented 
so when i cp'ed it from telegraf.local.sample telegraf.local i made no changes 
in the telgraf docker 
i see this 
```
2023-02-20T15:45:03Z E! error loading config file /etc/telegraf/telegraf.d/local.conf: open /etc/telegraf/telegraf.d/local.conf: permission denied
2023-02-20T15:46:07Z E! error loading config file /etc/telegraf/telegraf.d/local.conf: open /etc/telegraf/telegraf.d/local.conf: permission denied
```
and i cant login to it to check 
```
root@musafir-nas-718:/volume1/@docker/Powerwall-Dashboard# docker exec -it telegraf /bin/bash
Error response from daemon: Container 379493e78acc1497ee523e0ee61d527781a244474db81d9f2b61a83ee51b6f83 is restarting, wait until the container is running
```

In Powerwall.yml
i see if referring to telegraf.conf
```
 telegraf:
        image: telegraf:latest
        container_name: telegraf
        hostname: telegraf
        restart: always
        user: "1000:1000"
        command: [
            "telegraf",
            "--config",
            "/etc/telegraf/telegraf.conf",
            "--config-directory",
            "/etc/telegraf/telegraf.d"
        ]
```
Should it be telegraf.local ?

**Reply by jasonacox** [reply]
Thankfully, telegraf is very simple. It only need access to that one file.  This seems to be a permission issue.  Can you check?

```bash
# check permission on conf files
ls -l telegraf*
```

They should show "-rw-r--r--" but if they don't you can update them:

```bash
# change permission on conf files
chmod a+r telegraf*

# check telegraf
docker logs telegraf -f
```

You don't even need to restart telegraf as it keep trying to open the file.  Hopefully you will see something like this in the logs:

```
2023-02-20T16:01:07Z E! error loading config file /etc/telegraf/telegraf.conf: open /etc/telegraf/telegraf.conf: permission denied
2023-02-20T16:01:12Z E! error loading config file /etc/telegraf/telegraf.conf: open /etc/telegraf/telegraf.conf: permission denied
2023-02-20T16:01:19Z I! Starting Telegraf 1.25.2
2023-02-20T16:01:19Z I! Available plugins: 227 inputs, 9 aggregators, 26 processors, 21 parsers, 57 outputs, 2 secret-stores
2023-02-20T16:01:19Z I! Loaded inputs: cpu disk diskio http (2x) kernel mem processes swap system
2023-02-20T16:01:19Z I! Loaded aggregators: 
2023-02-20T16:01:19Z I! Loaded processors: date (2x)
2023-02-20T16:01:19Z I! Loaded secretstores: 
2023-02-20T16:01:19Z I! Loaded outputs: influxdb
2023-02-20T16:01:19Z I! Tags enabled: host=telegraf
2023-02-20T16:01:19Z I! [agent] Config: Interval:5s, Quiet:false, Hostname:"telegraf", Flush Interval:10s
```

**Reply by fuadar** [reply]
This was indeed the issue. i did have to stop and start it but i didn't realize (duh) the issue was not in the container but on my local fs which after reading your comments totally made sense and it works 
![image](https://user-images.githubusercontent.com/2882496/220158049-8b0f38ac-4853-404e-a51e-02a64a74ea89.png)

**Reply by jasonacox** [reply]
Awesome!  Congratulations @fuadar !

I would like to see if we need to adjust the manual steps to help others avoid the same pitfall you ran into.  Could you share details on the type of system you are running?  I ran the manual steps on a few of test systems (Ubuntu and Raspberry Pi) and couldn't reproduce a scenario where the file permission were not world readable. Do you know why the permission might have been set differently?

**Reply by fuadar** [reply]
oh absolutely so i'm running on a synology but not running the rootless step.
I git cloned the repo on my laptop. Made the changes then scp'ed the file as my user to the syno
sudo su - 
cd  /volume1/@docker/Powerwall-dashboard
cp /tmp/Powerwall-dashboard.tar .
tar xvf Powerwall-dashboard.tar
so i was working between my user and root and the cp i was doing sudo'ed as root on my synology was preserving the permissions of the orignal files which was the issue

**Reply by jasonacox** [reply]
Understood!  In general running as root breaks the installation which is why the setup.sh script won't run as root.  But your steps seem logical. I should add a note to the manual setup instruction to make sure ownership/permission on config files are set so that the docker services can read them.  Thanks for your help!
